app/models/nameserver.rb

```diff
@@ -88,7 +88,7 @@ def glue_record_required?
   end
 
   def normalize_attributes
-    self.hostname = hostname.try(:strip).try(:downcase)
+    self.hostname = hostname.try(:strip).try(:downcase).gsub(/\.$/, '')
```

Comment:
For some reason we were asserting :strip method to remove last dot from string. It is not, so for tests to be OK and to continue usual behaviour I've added :gsub.